File: components/redis-handling-session/src/main/java/com/sv/youapp/component/authorization/entities/redis/OAuth2TokenExchangeGrantAuthorization.java
package com.sv.youapp.component.authorization.entities.redis;

public class OAuth2TokenExchangeGrantAuthorization extends AbstractGrantAuthorization {

	public OAuth2TokenExchangeGrantAuthorization(
		String id,
		String registeredClientId,
		String principalName,
		String scopes,
		AbstractGrantAuthorization.AccessToken accessToken
	) {
		super(id, registeredClientId, principalName, scopes, accessToken, null);
	}
}

File: authorization-server/src/main/kotlin/com/sv/youapp/authorization/entities/redis/OAuth2TokenExchangeGrantAuthorization.kt
package com.sv.youapp.authorization.entities.redis

class OAuth2TokenExchangeGrantAuthorization(
    id: String?,
    registeredClientId: String?,
    principalName: String,
    scopes: String,
    accessToken: AccessToken?,
) : AbstractGrantAuthorization(id, registeredClientId, principalName, scopes, accessToken, null)
